Only skip specific failures on JRuby instead of ignoring the whole CI feedback

deivid-rodriguez opened the pull request:
This PR is just some experiment to proof that https://github.com/jruby/ruby-maven/pull/2 fixes:

* Issue reported at https://github.com/ruby/psych/issues/479#issuecomment-785469484, where jruby tests fail to even get started in CI (`bundle install` fails).
* https://github.com/ruby/psych/issues/520.
* https://github.com/jruby/ruby-maven/issues/3.
* https://github.com/rubygems/rubygems/issues/4685.

deivid-rodriguez commented:
Alright so dependent PRs are moving forward, and this is now only missing a `ruby-maven` release.

However, the gem became uninstallable again on `jruby` after https://github.com/ruby/psych/pull/525, since gemified `stringio` does not yet support jruby (WIP for that is https://github.com/ruby/stringio/pull/21).

headius commented:
This would be worth revisiting with latest SnakeYAML. On master, I now see 5F7E. Some of these may be valid differences (SnakeYAML appears to report line/column positioning a bit differently) but several appear to be fixable errors or bugs. We also addressed the stringio gemification for JRuby 9.4.

deivid-rodriguez commented:
@headius I completely lost context of this PR. I think the original issues I wanted to address are now fixed, but a few others remain. I changed the scope of this PR to:

* Use Bundler to install dependencies and run tests.
* Only skip the particular tests that are failing instead of ignoring the whole test run outcome. I think this should help preventing new regressions.

Let's see what CI says.

deivid-rodriguez commented:
Except for the specs that are marked as pending, everything is now green on JRuby, so I set this as ready for review.

headius commented:
@deivid-rodriguez Very nice! Good idea to skip some tests that are problematic... some might not be fixable due to minor differences in SnakeYAML (positioning information etc).

@tenderlove Any objections to merging this PR? It marks a number of tests as pending, but I'm not sure they are fixable and it would be good to have a green-ish run so we can see when things regress.

hsbt commented:
@deivid-rodriguez 👍 Thanks!
